repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -56,41 +102,113 @@ def parse_file(path):
         return [parse_entry(entry) for entry in data['packageinfo']]
 
 
-def filter_events(events):
-    """ Filter PES events """
-    installed_pkgs = set()
-    for pkgs in api.consume(InstalledRedHatSignedRPM):
-        installed_pkgs.update([pkg.name for pkg in pkgs.items])
+def parse_entry(entry):
+    """Parse PES event data"""
+    action = parse_action(entry['action'])
+    in_pkgs = parse_packageset(entry.get('in_packageset', {}) or {})
```

Comment:
Let me see. I'm not the author.

Reply:
Well not if the entry could be none, you only get the default value if the key does not exist

Reply:
1. PES currently does not produce events without 'in_packageset' or 'out_packageset'. If there are no "out" packages, PES gives you `'out_packageset': null` in JSON translating to  `'out_packageset': None` in Python. Meaning that `entry.get('out_packageset', {})` gives you `None` and not `{}`. And we don't want to pass `None` to the `parse_packageset()`.
1. Simplification: I changed it to `in_pkgs = parse_packageset(entry.get('in_packageset') or {})` because {} does the same job as the _get()_'s default `None` when using it with the shortcircuiting `or` operator.
   Note: We could also use `in_pkgs = parse_packageset(entry['out_packageset'] or {})` because the PES events currently always contain the `out_packageset` key, but you know, maybe it won't be like that in future and then it would crash with KeyError.

Reply:
Ahh I see, thanks for the clarification guys...